### DISCUSSION TITLE
Is there a way to (manually) trigger persistence mechanism?

### DISCUSSION THREAD

**Discussion by ingted** [NTDLS/Katzebase#94] [Q&A]
```
        use transactionReference = _core.Transactions.Acquire(preLogin)
#if GENERIC_TDATA
        let d = new KbInsensitiveDictionary<fstring>()
#if CELL_STRUCT
        let insert100Resp =
            [1..100]
            |> List.iter (fun i ->
                d.Add($"guy{i}", T ("ttc", A [| S "OGC"; D (double i) |]))
            )
            let result = new KbQueryDocumentStoreReply()
            let docPointer =
                _core.Documents.InsertDocument(
                    transactionReference.Transaction,
                    testSchemaIO,
                    (new KbDocument(d)).Content
                    )
            result.Value <- docPointer.DocumentId
            transactionReference.CommitAndApplyMetricsThenReturnResults(result, d.Count)
        equals 100 insert100Resp.RowCount
        transactionReference.Commit()
```

However when the fsi process ended, and the test io schema not dropped, the identical 100 records could be inserted again... (without key confliction)
Do I miss deferedIO stuff? What's the correct way to persist data? (I tried to do validation step-by-step for functionalities separately... )

**Comment by ingted** [comment]
PS. I know I am using the old version, but it takes time to catch up the latest commits... 
PS. The tests are now compatible to non-generic/generic versions.

**Comment by NTDLS** [comment]
I have some code suggestions, heading to my computer. What are you trying to do? Insert 100 rows twice? You won’t ever get any conflicts unless you add a unique key.

**Reply by ingted** [reply]
Oh... oops, I got it now. I was so stupid... = =||| 
It was that I inserted several rows, not, fields in a document. 
I am so sorry.... Let me close this.

**Comment by NTDLS** [comment]
Hahaha, nah. You’re awesome man. Also, I fell asleep. Sorry. 😁
